src/idls/erc_721.did.ts

```diff
@@ -0,0 +1,215 @@
+export default ({ IDL }) => {
+  const AccountIdentifier = IDL.Text;
+  const User = IDL.Variant({
+    principal: IDL.Principal,
+    address: AccountIdentifier,
+  });
+  const TokenIdentifier = IDL.Text;
+  const Memo = IDL.Vec(IDL.Nat8);
+  const SubAccount = IDL.Vec(IDL.Nat8);
+  const Balance = IDL.Nat;
+  const TransferRequest = IDL.Record({
```

Comment:
these are actually EXT data structures, the reference NFT supports both EXT and DIP721